lib/orderbook/OrderBook.ts

```diff
@@ -927,7 +934,13 @@ class OrderBook extends EventEmitter {
       this.swaps.on('swap.paid', paidHandler);
     }
 
-    return remainingQuantityToRemove;
+    const remainingFromOrder = orderQuantity - removedQuantity;
+    return {
+      removedQuantity,
+      onHoldQuantity,
+      pairId: order.pairId,
+      remainingQuantity: remainingFromOrder > onHoldQuantity ? remainingFromOrder : 0,
```

Comment:
This seems off to me. If we're trying to indicate the quantity that will remain on the order after any holds or complete, I figure it should be:

```typescript
const remainingQuantity = orderQuantity - removedQuantity - remainingQuantityToRemove;
```

Or perhaps simpler we just use the current quantity on the order (after we've removed whatever we were able to before holds) in which case I don't think we need the `orderQuantity` variable at all.

```typescript
const remainingQuantity = order.quantity - remainingQuantityToRemove;
```